### ISSUE TITLE
ArgumentNullException in EditorConfigCache

### ISSUE THREAD

**Issue by KirillOsenkov** [mono/monodevelop#5790]
This is in master as of August 28 2018.

When creating a new .html file there's a first-chance ArgumentNullException:

```

  | Microsoft.VisualStudio.CodingConventions.EditorConfig.EditorConfigCache+<GetConfigsAsync>d__14.MoveNext Line 266 |  
-- | -- | --
  | System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1.Start Line 471 |  
  | Microsoft.VisualStudio.CodingConventions.EditorConfig.EditorConfigCache.GetConfigsAsync Line |  
  | Microsoft.VisualStudio.CodingConventions.EditorConfig.EditorConfigCache+<GetCodingConventionSnapshotAsync>d__8.MoveNext Line 41 |  
  | System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1.Start Line 471 |  
  | Microsoft.VisualStudio.CodingConventions.EditorConfig.EditorConfigCache.GetCodingConventionSnapshotAsync Line |  
  | Microsoft.VisualStudio.CodingConventions.EditorConfig.EditorConfigCodingConventionsProvider+<GetCodingConventionsAsync>d__4.MoveNext Line 34 |  
  | System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1.Start Line 471 |  
  | Microsoft.VisualStudio.CodingConventions.EditorConfig.EditorConfigCodingConventionsProvider.GetCodingConventionsAsync Line |  
  | Microsoft.VisualStudio.CodingConventions.CodingConventionsManager+<GetConventionContextAsync>d__13.MoveNext Line 81 |  
  | System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1.Start Line 471 |  
  | Microsoft.VisualStudio.CodingConventions.CodingConventionsManager.GetConventionContextAsync Line |  
  | MonoDevelop.Ide.Editor.EditorConfigService.GetEditorConfigContext Line 52 |  
  | MonoDevelop.Ide.Editor.TextEditorViewContent+<UpdateStyleParent>d__15.MoveNext Line 167 |  
  | System.Runtime.CompilerServices.AsyncTaskMethodBuilder.Start Line 316 |  
  | MonoDevelop.Ide.Editor.TextEditorViewContent.UpdateStyleParent Line |  
  | MonoDevelop.Ide.Editor.TextEditorViewContent.UpdateTextEditorOptions Line 115 |  
  | MonoDevelop.Ide.Editor.TextEditorViewContent.OnSetProject Line 267 |  
  | MonoDevelop.Ide.Gui.BaseViewContent.set_Project Line 132 |  
  | MonoDevelop.Ide.Editor.TextEditorDisplayBinding.CreateContent Line 107 |  
  | MonoDevelop.Ide.Gui.Workbench.NewDocument Line 722 |  
  | MonoDevelop.Ide.Templates.FileTemplate+<CreateFile>d__71.MoveNext Line 353 |  
  | System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1.Start Line 471 |  
  | MonoDevelop.Ide.Templates.FileTemplate.CreateFile Line |  
  | MonoDevelop.Ide.Templates.FileTemplate+<Create>d__67.MoveNext Line 255 |  
  | System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1.Start Line 471 |  
  | MonoDevelop.Ide.Templates.FileTemplate.Create Line |  
  | MonoDevelop.Ide.Projects.NewFileDialog+<OpenEvent>d__38.MoveNext Line 537 |  
  | System.Runtime.CompilerServices.AsyncVoidMethodBuilder.Start Line 84 |  
  | MonoDevelop.Ide.Projects.NewFileDialog.OpenEvent Line |  


```

> VS bug [#673075](https://devdiv.visualstudio.com/DevDiv/_workitems/edit/673075)

**Comment by KirillOsenkov** [comment]
This try/catch does nothing as the method is not async:
http://source.monodevelop.com/#MonoDevelop.Ide/MonoDevelop.Ide.Editor/EditorConfigService.cs,54

**Comment by KirillOsenkov** [comment]
This invocation is inside async void, and doesn't catch the exception:
http://source.monodevelop.com/#MonoDevelop.Ide/MonoDevelop.Ide.Editor/TextEditor.cs,1069

**Comment by xamarin-release-manager** [comment]
Fixed in version 7.8.0.570 (master)

Pull Request #6049 merged by: Mike Krüger
Author: mono
Commit: c5b2b6cd9eaa12365afdb783368a3c4f64f695dd (mono/monodevelop)

**Comment by mrward** [comment]
Just had the ArgumentNullException, in 7.7.0.1424 which should include the fix. Seems to happen for a new file (e.g. xml) that has not been saved. My repro steps:

With a project open, File - New File, select XML file, uncheck Add to project, click New. 

Not sure what the code is doing. Does it somewhere do a Path.GetDirectory and the filename does not have a path? I think a new unsaved file has a name but not a full path.

Going to re-open this. If we think this is not the same bug I could create a new GitHub issue?

```
Error while getting coding conventions,
System.ArgumentNullException: Value cannot be null.
Parameter name: dir
  at Microsoft.VisualStudio.CodingConventions.EditorConfig.EditorConfigCache+<GetConfigsAsync>d__14.MoveNext () [0x0002b] in <33418e5da6fc40388dc1fe4aee49e148>:0 
--- End of stack trace from previous location where exception was thrown ---
  at Microsoft.VisualStudio.CodingConventions.EditorConfig.EditorConfigCache+<GetCodingConventionSnapshotAsync>d__8.MoveNext () [0x0009e] in <33418e5da6fc40388dc1fe4aee49e148>:0 
--- End of stack trace from previous location where exception was thrown ---
  at Microsoft.VisualStudio.CodingConventions.EditorConfig.EditorConfigCodingConventionsProvider+<GetCodingConventionsAsync>d__4.MoveNext () [0x00099] in <33418e5da6fc40388dc1fe4aee49e148>:0 
--- End of stack trace from previous location where exception was thrown ---
  at Microsoft.VisualStudio.CodingConventions.CodingConventionsManager+<GetConventionContextAsync>d__13.MoveNext () [0x0012b] in <33418e5da6fc40388dc1fe4aee49e148>:0 
--- End of stack trace from previous location where exception was thrown ---
  at MonoDevelop.Ide.Editor.EditorConfigService+<GetEditorConfigContext>d__4.MoveNext () [0x0005a] in /Users/vsts/agent/2.140.2/work/1/s/monodevelop/main/src/core/MonoDevelop.Ide/MonoDevelop.Ide.Editor/EditorConfigService.cs:51 
```

Build Information
Release ID: 707001424
Git revision: 2fefd1804b10c65136993614dc0afb3f197903d8
Build date: 2018-09-25 21:06:25+00
Build branch: release-7.7
Xamarin extensions: 8c052293f9d925b55a5596022f1fd61e02601036

**Comment by mkrueger** [comment]
No it's not the same but similiar.

**Comment by xamarin-release-manager** [comment]
Fixed in version 7.8.0.606 (master)

Pull Request #6099 merged by: Lluis Sanchez
Author: mono
Commit: c4869486da1636056435452236bd6b7d5554ff84 (mono/monodevelop)

**Comment by xamarin-release-manager** [comment]
Fixed in version 8.0.0.48 (release-8.0)

Pull Request #6099 merged by: Lluis Sanchez
Author: mono
Commit: c4869486da1636056435452236bd6b7d5554ff84 (mono/monodevelop)

**Comment by xamarin-release-manager** [comment]
Fixed in version 7.7.0.1462 (release-7.7)

Pull Request #6100 merged by: Lluis Sanchez
Author: mono
Commit: eb744660e678c7c30b31f0582b4c102bc5f99934 (mono/monodevelop)

**Comment by xamarin-release-manager** [comment]
Fixed in version 7.7.0.1462 (release-7.7)

Pull Request #6100 merged by: Lluis Sanchez
Author: mono
Commit: eb744660e678c7c30b31f0582b4c102bc5f99934 (mono/monodevelop)

**Comment by xamarin-release-manager** [comment]
Fixed in version 8.6.0.802 (master)

Pull Request #6099 merged by: Lluis Sanchez
Author: mono
Commit: c4869486da1636056435452236bd6b7d5554ff84 (xamarin/monodevelop)
